File: packages/agent-client/typescript/src/stream.ts
import type { RawStream, Stream } from "@microsandbox/protocol-client";
import type { AgentProtocol } from "./protocol.js";

/** Owned agent message subscription, with native sends and optional split ownership. */
export type AgentStream = Stream<AgentProtocol>;
/** Owned opaque agent frame subscription. */
export type RawAgentStream = RawStream<AgentProtocol>;
export { Stream, RawStream, StreamSender, StreamReceiver, RawStreamSender, RawStreamReceiver } from "@microsandbox/protocol-client";

File: packages/agent-client/rust/lib/stream.rs
//! Agent streams backed by the actual shared router.

//--------------------------------------------------------------------------------------------------
// Re-Exports
//--------------------------------------------------------------------------------------------------

pub use microsandbox_protocol_client::{
    RawStream, RawStreamReceiver, RawStreamSender, Stream, StreamReceiver, StreamSender,
};

/// Owned decoded agent stream.
pub type AgentStream = Stream<crate::AgentProtocol>;
/// Owned raw agent stream.
pub type RawAgentStream = RawStream<crate::AgentProtocol>;
